docs/excel/pnp-open-in-excel.md

```diff
@@ -0,0 +1,200 @@
+1. Add a new scope Files.ReadWrite to the ida:AppScopes key so that it looks like the following code:
+    ```xml
+    <add key="ida:AppScopes" value="User.Read Calendars.Read Files.ReadWrite " />
+    ```
+1. Open the Index.cshtml file.
+1. Insert the following ActionLink code to create a button for uploading a file to OneDrive.
+    ```razor
+    @if (Request.IsAuthenticated)
+    {
+        <h4>Welcome @ViewBag.User.DisplayName!</h4>
+        <p>Use the navigation bar at the top of the page to get started.</p>
+        @Html.ActionLink("Click here to create a new file on OneDrive", "CreateOneDriveFile", "Home", new { area = "" }, new { @class = "btn btn-primary btn-large" })
+    }
+    ```
+1. Open the HomeController.cs file.
+1. Insert the following code to handle the request from the action link.
+    ```csharp
+    public void CreateOneDriveFile()
+        {
+            using (var stream = new System.IO.MemoryStream(System.Text.Encoding.UTF8.GetBytes("The contents of the file goes here.")))
+            {
+                var client = graph_tutorial.Helpers.GraphHelper.UploadFile("/test.txt", stream);
+            }
+        }
+    ```
+1. Open the GraphHelper.cs file.
+1. Insert the following code to call the Microsoft Graph API to create a new file on OneDrive.
+    ```csharp
+    public static async Task UploadFile(string fileName, System.IO.MemoryStream stream)
+        {
+           var graphClient = GetAuthenticatedClient();
+            await graphClient.Me
+                .Drive
+                .Root
+                .ItemWithPath(fileName)
+                .Content
+                .Request()
+                .PutAsync<DriveItem>(stream);
+            return;
+        }
+    ```
+1. Press F5 (or Debug > Start Debugging). The web application will start.
+1. Choose Click here to sign in, and sign in.
+1. Choose Click here to create a new file on OneDrive.
+1. Open a new browser tab and sign in to your OneDrive account. You will see the test.txt file in the root folder.
+
+Now that you have seen how to upload a file to OneDrive, you can reuse this code to upload any Excel document that you create.
+
+## Additional considerations for your solution
+
+Everyone’s solution is different in terms of technologies and approaches. The following considerations will help you when planning out how to modify your solution to open documents and embed your Office add-in.
+
+### Create a new Excel spreadsheet from the web page
+
+The sample modifies an existing Excel document. A more common scenario is that you’ll create a new Excel spreadsheet from your web page. You can find additional details on how to create a new spreadsheet in Create a spreadsheet document by providing a file name. This article shows how to create the file locally, but you can also create the file in a stream by using an overload on the SpreadsheetDocument.Create method.
```

Comment:
```suggestion
The sample modifies an existing Excel document. A more common scenario is that you’ll create a new Excel spreadsheet from your web page. You can find additional details on how to create a new spreadsheet in **Create a spreadsheet document** by providing a file name. This article shows how to create the file locally, but you can also create the file in a stream by using an overload on the SpreadsheetDocument.Create method.
```
This is a bit confusing, is "Create a spreadsheet document" a link to a topic or a UI feature? And what is "This article?" Please clarify using either a link to the article or define the UI more clearly.